nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/upload/BatchUpload.java

```diff
@@ -83,50 +81,34 @@ public BatchUpload createBatch() {
         return (BatchUpload) getResponse();
     }
 
-    protected Object upload(String fileName, long fileSize, String fileType, String uploadType,
-            String uploadChunkIndex, String totalChunkCount, String batchId, String fileIdx, File file) {
-        RequestBody fbody = RequestBody.create(MediaType.parse(fileType), file);
-        return getResponse(fileName, Objects.toString(fileSize), fileType, uploadType, uploadChunkIndex,
-                totalChunkCount, batchId, fileIdx, fbody);
-    }
-
     public BatchUpload upload(String name, long length, String fileType, String batchId, String fileIdx, File file) {
         if (chunkSize == 0) {
-            return (BatchUpload) upload(name, length, fileType, ConstantsV1.UPLOAD_NORMAL_TYPE, "0", "1", batchId,
-                    fileIdx, file);
+            // Post file
+            RequestBody fbody = RequestBody.create(MediaType.parse(fileType), file);
+            return (BatchUpload) getResponse(name, Objects.toString(length), fileType, ConstantsV1.UPLOAD_NORMAL_TYPE,
+                    "0", "1", batchId, fileIdx, fbody);
         }
-        int partCounter = 1;
-        int sizeOfFiles = chunkSize;
-        List<File> files = new ArrayList<>();
-        BatchUpload batchUpload = null;
-        byte[] buffer = new byte[sizeOfFiles];
-        int tmp = 0;
-        try {
-            try (BufferedInputStream bis = new BufferedInputStream(new FileInputStream(file))) {
-                while ((tmp = bis.read(buffer)) > 0) {
-                    // write each chunk of data into separate file with different number in name
-                    File newFile = new File(file.getParent(), name + "." + String.format("%03d", partCounter++));
-                    try (FileOutputStream out = new FileOutputStream(newFile)) {
-                        out.write(buffer, 0, tmp); // tmp is chunk size
-                        files.add(newFile);
-                    }
-                }
+        try (BufferedInputStream bis = new BufferedInputStream(new FileInputStream(file))) {
+            Object response = null;
+            int contentLength = 0;
+            byte[] buffer = new byte[chunkSize];
```

Comment:
Using a `MAXIMUM_CHUNK_SIZE` for instance? What would it be then?
And in this case it should probably be handled in the
```
    public BatchUpload chunkSize(int chunkSize) {
        this.chunkSize = chunkSize;
        return this;
    }
```
method, because by default when enbaling chunks the size is set to 1024 * 1024: 
```
    public BatchUpload enableChunk() {
        this.chunkSize = ConstantsV1.CHUNK_SIZE;
        return this;
    }
```